src/SolverBenchmark.jl

```diff
@@ -9,6 +9,7 @@ using JLD2
 using PrettyTables
 
 using NLPModels
+using JuMP, NLPModelsJuMP
```

Comment:
Hi @paraynaud ! Thanks for the PR. Though, I am not too fond of having `JuMP` and `NLPModelsJuMP` as dependencies here.
What is the motivation for it?

Reply:
Hi @tmigot !
This PR is related to performance profiles considering large partially-separable models (`n>10000`).
I found that an MathOptNLPModel retrieves a lot faster the expression tree of the objective than an ADNLPModel.
However, I had an `OutOfMemory` error when I tried to allocate every MathOptNLPModels before running `bmark_solver` and `bmark_solver` doesn't support `JuMP` models.
This PR remedy to this issue, by allocating on the fly the MathOptNLPModels from the JuMP.Model (in `solve_problems`).

Originally, I didn't intend to open a PR on JSO.
I only wanted to open this PR on my forked repo, but I didn't check the branch on which it would be merged and by default it was the one from JSO.
I left it open, because I though it might interest others, but I don't need it to be merged, having the branch is enough to run the modified `bmark_solver` on an server.

Reply:
I understand the memory issue. Usually this work (source: https://juliasmoothoptimizers.github.io/OptimizationProblems.jl/dev/benchmark/ ):
```
problems = (MathOptNLPModel(eval(Meta.parse(problem))(), name=problem) for problem ∈ OptimizationProblems.meta[!, :name])
```
by using `()` instead of `[]` you create an `Iterator` that does lazy evaluations of the "array" so you don't allocate all the models first.

Reply:
Many thanks!! I'll try this soon.
I'll close this PR now.